rotkehlchen/globaldb/handler.py

```diff
@@ -439,10 +440,27 @@ def retrieve_assets(userdb: 'DBHandler', filter_query: 'AssetsFilterQuery') -> t
                     UnderlyingToken.deserialize_from_db((entry[1], entry[2], entry[3])).serialize(),  # noqa: E501
                 )
 
-            # get `entries_found`
+            # get `entries_found`. In the case of handling the ignored assets we need to manually
+            # count the assets found
             query, bindings = filter_query.prepare(with_pagination=False)
-            total_found_query = f'SELECT COUNT(*) FROM ({parent_query}) ' + query
-            entries_found = cursor.execute(total_found_query, bindings).fetchone()[0]
+            if filter_query.ignored_assets_handling == IgnoredAssetsHandling.NONE:
+                total_found_query = f'SELECT COUNT(*) FROM ({parent_query}) ' + query
+                entries_found = cursor.execute(total_found_query, bindings).fetchone()[0]
+            else:
+                total_found_query = f'SELECT identifier FROM ({parent_query}) ' + query
```

Comment:
Why do it like this? Why not just do modify the sqlite query?

Reply:
We don't do it purely in sqlite because we need the ignored assets that are in the user db